index.php

```diff
@@ -124,8 +124,17 @@
 		}
 		elseif(my_strpos($user['sid'], 'bot=') !== false && $spiders[$botkey])
 		{
+			if($mybb->settings['wolorder'] == 'username')
+			{
+				$key = $spiders[$botkey]['name'];
+			}
+			else
+			{
+				$key = $user['time'];
+			}
+
 			// The user is a search bot.
-			$onlinebots[] = format_name($spiders[$botkey]['name'], $spiders[$botkey]['usergroup']);
+			$onlinebots[$key] = format_name($spiders[$botkey]['name'], $spiders[$botkey]['usergroup']);
 			++$botcount;
```

Comment:
Also here why key even matter if I loop through already sorted result and assign (auto) numeric keys?